### PR TITLE
Feature/17 1 implement methods for creating time slot

### PR DESCRIPTION
**Pull request by eugent482**

### CODE REVIEW

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using InternetHospital.DataAccess.Entities;
+using System;
+using System.Linq;
+
+namespace InternetHospital.BusinessLogic.Services
+{
+    public class AppointmentService : IAppointmentService
+    {
+        private readonly ApplicationContext _context;
+        const int MINIMUM_APPOINTMENT_TIME = 15;
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: If you explicitly used private access level, you should do it everywhere.

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        /// <summary>
+        /// Get all open and reserved appointments for current doctor
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
+        public IQueryable<AppointmentModel> GetMyAppointments(int doctorId)
+        {
+            var appointments = _context.Appointments
+                .Where(x => (x.DoctorId == doctorId)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: Better to rename x to a, as Appointment

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                return (false, $"Your appointment has to be at least {MINIMUM_APPOINTMENT_TIME} minutes");
+            }
+            if (creationModel.StartTime.Date != creationModel.EndTime.Date)
+            {
+                return (false, "Your appointment can't be bigger then one day");
+            }
+            var isUnique = _context.Appointments.Any(x => x.DoctorId == doctorId
+                                                       && ((creationModel.StartTime >= x.StartTime && creationModel.StartTime < x.EndTime)
+                                                           || (creationModel.EndTime > x.StartTime && creationModel.EndTime <= x.EndTime)));
+            if (isUnique)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:73]: You used {} everywhere, i think you should add them here

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        /// <summary>
+        /// Get all available appointments of selected doctor 
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
+        public IQueryable<AvailableAppointmentModel> GetAvailableAppointments(int doctorId)
+        {
+            var appointments = _context.Appointments
+                .Where(x => (x.DoctorId == doctorId) && (x.StatusId == DEFAULT_STATUS))
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: x to a, as Appointment

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+        private bool CreateAppointment(AppointmentCreationModel model, int id)
+        {
+            try
+            {
+                var appointment = Mapper.Map<Appointment>(model);
+                appointment.StatusId = DEFAULT_STATUS;
+                appointment.DoctorId = id;
+                if (model.Address == null)
+                    appointment.Address = _context.Doctors.FirstOrDefault(x => x.UserId == id)?.Address;
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: x to d, as Doctor

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            }
+            if ((creationModel.EndTime - creationModel.StartTime).TotalMinutes < MINIMUM_APPOINTMENT_TIME)
+            {
+                return (false, $"Your appointment has to be at least {MINIMUM_APPOINTMENT_TIME} minutes");
+            }
+            if (creationModel.StartTime.Date != creationModel.EndTime.Date)
+            {
+                return (false, "Your appointment can't be bigger then one day");
+            }
+            var isUnique = _context.Appointments.Any(x => x.DoctorId == doctorId
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: x to a, as Appointment

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        /// Get all open and reserved appointments for current doctor
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
+        public IQueryable<AppointmentModel> GetMyAppointments(int doctorId)
+        {
+            var appointments = _context.Appointments
+                .Where(x => (x.DoctorId == doctorId) 
+                            && (x.StatusId == DEFAULT_STATUS || x.StatusId == RESERVED_STATUS))
+                .Select(x => new AppointmentModel
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs]: x to a, as Appointment

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 8]
```diff
@@ -17,24 +17,24 @@ public DoctorService(ApplicationContext context)
 
         public (IQueryable<DoctorModel> doctors, int count) GetAll(DoctorSearchParameters queryParameters)
         {
-            var _doctors = _context.Doctors.AsQueryable();
+            var doctors = _context.Doctors.Where(x=>x.IsApproved==true).AsQueryable();
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: x to d, as Doctor

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 9]
```diff
@@ -17,24 +17,24 @@ public DoctorService(ApplicationContext context)
...
 
             if (queryParameters.SearchByName != null)
             {
                 var toLowerSearchParameter = queryParameters.SearchByName.ToLower();
-                _doctors = _doctors
+                doctors = doctors
                     .Where(x => x.User.FirstName.ToLower().Contains(toLowerSearchParameter)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: x to d, as Doctor

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 10]
```diff
@@ -17,24 +17,24 @@ public DoctorService(ApplicationContext context)
...
                     || x.User.SecondName.ToLower().Contains(toLowerSearchParameter)
                     || x.User.ThirdName.ToLower().Contains(toLowerSearchParameter));
             }
 
             if (queryParameters.SearchBySpecialization != null)
             {
-                _doctors = _doctors.Where(x => x.SpecializationId == queryParameters.SearchBySpecialization);
+                doctors = doctors.Where(x => x.SpecializationId == queryParameters.SearchBySpecialization);
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: x to d, as Doctor

File: InternetHospital.BusinessLogic/Services/TokenService.cs [hunk 11]
```diff
@@ -18,8 +18,8 @@ namespace InternetHospital.BusinessLogic.Services
     public class TokenService : ITokenService
     {
         private readonly AppSettings _appSettings;
-        private ApplicationContext _context;
-        private UserManager<User> _userManager;
+        private readonly ApplicationContext _context;
+        private readonly UserManager<User> _userManager;
         const string APPROVED = "Approved";
         const string DOCTOR = "Doctor";
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/TokenService.cs]: i think it`s better to mark them explicitly private

**Review by SergiyKozachok — APPROVED** [review]

**Review by TREN-D — APPROVED** [review]

**Review by serhiidotsyk — APPROVED** [review]

**Pull request merged**